src/upnp.cpp

```diff
@@ -898,13 +889,18 @@ TORRENT_EXTRA_EXPORT void find_control_url(int type, char const* string, parse_s
 	else if (type == xml_string)
 	{
 		if (state.tag_stack.empty()) return;
-//		std::cout << " " << string << std::endl;
+//		std::cout << " " << string << std::endl;}
 		if (!state.in_service && state.top_tags("service", "servicetype"))
 		{
-			if (string_equal_no_case(string, state.service_type))
+			if (string_equal_no_case(string, "urn:schemas-upnp-org:service:WANIPConnection:1")
```

Comment:
> this is a great change. thanks! Do you think there's a simple way of creating a bias towards version 2? or does that even make sense?

Hmmm, I think there is no special need to favor version 2. Just supporting it is enough.

> The case I'm thinking of is if a router provides multiple services, right now we would pick the first one, previously we would prefer the WANIPConnection over PPPConnection.

I don't have much knowledge on this, but I think it will work fine. `WANIPConnection` is likely to be implemented on all IGDs:
```
2.2. Device Model Table 1
WANIPConnection
Required for modems that support IPv4 based connections
```